lib/client/api.go

```diff
@@ -3398,14 +3398,14 @@ func (tc *TeleportClient) DeviceLogin(ctx context.Context, certs *devicepb.UserC
 	}
 
 	// Allow tests to override the default authn function.
-	runCeremony := tc.dtAuthnRunCeremony
-	if runCeremony == nil {
-		runCeremony = dtauthn.RunCeremony
+	ceremony := tc.dtAuthnCeremony
+	if ceremony == nil {
+		ceremony = dtauthn.NewCeremony()
```

Comment:
As suggested above, keep all of this unchanged and do:

```go
		runCeremony = dtauthn.NewCeremony().Run
```

We don't really gain anything with the new interface and `mockCeremony`.